docs/man/hilbertspace.1
'\" t
.\"     Title: hilbertspace
.\"    Author: :doctype:manpage
.\" Generator: DocBook XSL Stylesheets v1.79.1 <http://docbook.sf.net/>
.\"      Date: 07/18/2016
.\"    Manual: System administration commands
.\"    Source: hilbertspace 0.0.2
.\"  Language: English
.\"
.TH "HILBERTSPACE" "1" "07/18/2016" "HilbertSpace 2\&.2\&.0" "System administration commands"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
hilbertspace \- REST based system administration server
.SH "SYNOPSIS"
.sp
hilbertspace [\-setup] [\-secrets] [\-hash]
.SH "DESCRIPTION"
.sp
\fBHilbertSpace\fR is an extra\-simple tool/framework/API for doing \*(Aqremote things\*(Aq\&.
.SH "ARGUMENTS"
.PP
\fBhost\-pattern\fR
.RS 4
A name of a group in the inventory, a shell\-like glob selecting hosts in inventory or any combination of the two separated by commas\&.
.RE
.SH "OPTIONS"
.PP
\fB\-setup\fR
.RS 4
Start guided initial setup of the application and users\&.
.RE
.PP
\fB\-secrets\fR
.RS 4
Generate cookie secrets\&.
.RE
.PP
\fB\-hash\fR
.RS 4
Generate hash of given password\&.
.RE
.PP
\fB\-\-version\fR
.RS 4
Show program version number and exit\&.
.RE
.SH "FILES"
.sp
/etc/hilbertspace.conf \(em Default configuration file
.sp
.SH "AUTHOR"
.sp
Hilbert Space was originally written by Gamunu Balagalla\&. See the AUTHORS file for a complete list of contributors\&.
.SH "COPYRIGHT"
.sp
Copyright \(co 2016, Gamunu Balagalla HilbertSpace is released under the terms of the MIT License\&.
.SH "SEE ALSO"
.sp
\fBansible\fR(1), \fBansible\-playbook\fR(1)
.sp
Extensive documentation is available in the documentation site: https://gamunu\&.github\&.io/hilbertspace\&. IRC and mailing list info can be found in file CONTRIBUTING\&.md
.SH "AUTHOR"
.PP
\fB:doctype:manpage\fR
.RS 4
Author.
.RE


CHANGELOG.md
## v2.0.2 | 23-5-2016

using semaphore branch


packaging/plugins/ansible/hilbertspace.py
# Make coding more python3-ish
from __future__ import (absolute_import, division, print_function)

__metaclass__ = type

import os
import yaml
from pymongo import MongoClient
from bson.objectid import ObjectId

from ansible.plugins.callback import CallbackBase


class CallbackModule(CallbackBase):
    """
    required "sudo pip install pymongo"

    logs ansible-playbook and ansible runs to a syslog server in json format
    make sure you have in ansible.cfg:
        callback_plugins   = <path_to_callback_plugins_folder>
    and put the plugin in <path_to_callback_plugins_folder>

    This plugin makes use of the following environment variables:
        HS_TASK_ID   (must)
        HS_TASK_TYPE     (optional): defaults to 1
    """
    CALLBACK_VERSION = 2.0
    CALLBACK_TYPE = 'aggregate'
    CALLBACK_NAME = 'hilbertspace'
    CALLBACK_NEEDS_WHITELIST = True

    def __init__(self):

        super(CallbackModule, self).__init__()

        # Get configuration object from hilbertspace
        # configuration file
        with open("/etc/hilbertspace.conf", 'r') as stream:
            try:
                self.config = yaml.load(stream)
            except yaml.YAMLError as exc:
                self.display.warning('Could not parse configuration file')

        # Build connection string and connect
        connection_str = "mongodb://{}/".format(",".join(self.config['mongodb']['hosts']))

        if len(self.config['mongodb']['replica_set']):
            connection_str += "?replicaSet={}".format(self.config['mongodb']['replica_set'])

        mongoclient = MongoClient(connection_str)

        # Get database
        self.db = mongoclient[self.config['mongodb']['name']]
        # authentication using configuration file
        self.db.authenticate(self.config['mongodb']['user'], self.config['mongodb']['pass'])

        self.taskID = ObjectId(os.getenv('HS_TASK_ID'))
        self.taskType = os.getenv('HS_TASK_TYPE', 1)

    def runner_on_failed(self, host, res, ignore_errors=False):
        result = self.db.addhoc_tasks.update_one(
            {"_id": self.taskID},
            {
                "$push": {
                    "task_status": {
                        "execution": "FAILED",
                        "host": host,
                        "message": res
                    }
                }
            }
        )

    def runner_on_ok(self, host, res):
        result = self.db.addhoc_tasks.update_one(
            {"_id": self.taskID},
            {
                "$push": {
                    "task_status": {
                        "execution": "SUCCESS",
                        "host": host,
                        "message": res
                    }
                }
            }
        )

    def runner_on_skipped(self, host, item=None):
        result = self.db.addhoc_tasks.update_one(
            {"_id": self.taskID},
            {
                "$push": {
                    "task_status": {
                        "execution": "SKIPPED",
                        "host": host
                    }
                }
            }
        )

    def runner_on_unreachable(self, host, res):
        result = self.db.addhoc_tasks.update_one(
            {"_id": self.taskID},
            {
                "$push": {
                    "task_status": {
                        "execution": "UNREACHABLE",
                        "host": host,
                        "message": res
                    }
                }
            }
        )

    def runner_on_async_failed(self, host, res):
        result = self.db.addhoc_tasks.update_one(
            {"_id": self.taskID},
            {
                "$push": {
                    "task_status": {
                        "execution": "ASYNC_FAILED",
                        "host": host,
                        "message": res
                    }
                }
            }
        )
